src/mbgl/renderer/render_source.cpp

```diff
@@ -64,12 +64,39 @@ void RenderSource::setObserver(RenderSourceObserver* observer_) {
 
 void RenderSource::onTileChanged(Tile& tile) {
     observer->onTileChanged(*this, tile.id);
+    if (tile.isComplete()) {
+        const auto tileID = tile.id.toUnwrapped();
+        const auto& idealTile = idealTilesLoaded.find(tileID);
+        if (idealTile != idealTilesLoaded.end()) {
+            idealTilesLoaded[tileID] = true;
+        }
+        for (const auto& entry : idealTilesLoaded) {
```

Comment:
here, we could just compare the size of `idealTilesLoaded` with the size of the actual ideal tiles from the tile pyramid (pls see the comments below)

Reply:
Done